### PR TITLE
[Admin][Product][Attribute] Fix adding existing attributes to product bug

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

Bug caused some nasty errors on ajax request.

zrzut ekranu 2016-12-05 o 15 53 45
zrzut ekranu 2016-12-05 o 15 54 41

### CODE REVIEW

File: features/product/managing_products/editing_product_attributes.feature [hunk 1]
```diff
@@ -0,0 +1,25 @@
+@managing_products
+Feature: Editing product's attributes
+    In order to modify product details
+    As an Administrator
+    I want to be able to edit product attributes
```
Comment (tuka217) [inline features/product/managing_products/editing_product_attributes.feature]: `product's`

File: src/Sylius/Behat/Page/Admin/Product/UpdateSimpleProductPage.php [hunk 2]
```diff
@@ -375,6 +403,8 @@ protected function getDefinedElements()
             'association_dropdown_item' => '.field > label:contains("%association%") ~ .product-select > div.menu > div.item:contains("%item%")',
             'association_dropdown_item_selected' => '.field > label:contains("%association%") ~ .product-select > a.label:contains("%item%")',
             'attribute' => '.attribute .label:contains("%attribute%") ~ input',
+            'attribute_delete_button' => '.attribute .label:contains("%attribute%") ~ button',
+            'attributes_popup_message' => '.popup .header',
```
Comment (pjedrzejewski) [inline src/Sylius/Behat/Page/Admin/Product/UpdateSimpleProductPage.php]: This selector is too generic.

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Attribute/attributeChoice.html.twig [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <div class="ui fluid action input" id="attributeChoice" data-action="{{ path('sylius_admin_render_attribute_forms') }}" style="margin-bottom: 15px;">
     {{ form_widget(form, {'attr': {'class': 'ui fluid search dropdown', 'id': 'sylius_product_attribute_choice'}}) }}
-    <button class="ui green labeled icon button">
+    <button id="addAttributesButton" class="ui green labeled icon button" data-title="{{ 'sylius.ui.no_new_attributes_selected'|trans }}">
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Attribute/attributeChoice.html.twig]: We should prefix these ids with ``sylius-`` = ``sylius-product-attributes-add-button`` or something like that?

File: src/Sylius/Bundle/UiBundle/Resources/private/js/sylius-product-attributes.js [hunk 4]
```diff
@@ -84,6 +97,12 @@
                 data: data,
                 dataType: 'html'
             }).done(function(data) {
+                if ('No new data' == data) {
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/UiBundle/Resources/private/js/sylius-product-attributes.js]: This looks hacky.

File: src/Sylius/Bundle/ProductBundle/Controller/ProductAttributeController.php [hunk 5]
```diff
@@ -75,6 +75,11 @@ public function renderAttributeValueFormsAction(Request $request)
         $form->handleRequest($request);
 
         $attributes = $form->getData();
+
+        if (null === $attributes) {
+            return new Response('No new data');
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ProductBundle/Controller/ProductAttributeController.php]: Not a good solution, why can't we handle it and return appropriate code here?

**Comment by pjedrzejewski** [comment]
Thank you Mateusz!

**Pull request merged**